tests/utils.py

```diff
@@ -276,6 +277,24 @@ def create_scalars_tbl(num_rows: int, seed: int = 0, percent_nulls: int = 10) ->
     return tbl
 
 
+def create_arrays(
```

Comment:
I'd all this `infinite_array_iterator` or some such to make it really clear what it does (it's an unusual pattern)